From a61910fde100fa1112ec57dd3e456719724997ef Mon Sep 17 00:00:00 2001
From: Carl George <carl@george.computer>
Date: Wed, 9 Feb 2022 17:15:39 -0600
Subject: [PATCH] Switch from nose to pytest

This is not a full rewrite to pytest style tests, it just changes the
minimum to allow pytest to run the existing tests.

Resolves #72
---
 .github/workflows/pythonpackage.yml | 6 +++---
 Makefile                            | 3 +--
 README.rst                          | 4 ++--
 gitdb.pro.user                      | 3 +--
 gitdb/test/db/test_pack.py          | 4 ++--
 gitdb/test/lib.py                   | 4 ++--
 gitdb/test/test_pack.py             | 4 ++--
 7 files changed, 13 insertions(+), 15 deletions(-)

diff --git a/gitdb/test/db/test_pack.py b/gitdb/test/db/test_pack.py
index ff96a58..4539f42 100644
--- a/gitdb/test/db/test_pack.py
+++ b/gitdb/test/db/test_pack.py
@@ -16,7 +16,7 @@
 import random
 import sys
 
-from nose.plugins.skip import SkipTest
+import pytest
 
 class TestPackDB(TestDBBase):
 
@@ -24,7 +24,7 @@ class TestPackDB(TestDBBase):
     @with_packs_rw
     def test_writing(self, path):
         if sys.platform == "win32":
-            raise SkipTest("FIXME: Currently fail on windows")
+            pytest.skip("FIXME: Currently fail on windows")
 
         pdb = PackedDB(path)
 
diff --git a/gitdb/test/lib.py b/gitdb/test/lib.py
index a04084f..abd4ad5 100644
--- a/gitdb/test/lib.py
+++ b/gitdb/test/lib.py
@@ -65,8 +65,8 @@ def skip_on_travis_ci(func):
     @wraps(func)
     def wrapper(self, *args, **kwargs):
         if 'TRAVIS' in os.environ:
-            import nose
-            raise nose.SkipTest("Cannot run on travis-ci")
+            import pytest
+            pytest.skip("Cannot run on travis-ci")
         # end check for travis ci
         return func(self, *args, **kwargs)
     # end wrapper
diff --git a/gitdb/test/test_pack.py b/gitdb/test/test_pack.py
index 4b01741..f946197 100644
--- a/gitdb/test/test_pack.py
+++ b/gitdb/test/test_pack.py
@@ -26,7 +26,7 @@
 from gitdb.exc import UnsupportedOperation
 from gitdb.util import to_bin_sha
 
-from nose import SkipTest
+import pytest
 
 import os
 import tempfile
@@ -246,4 +246,4 @@ def rewind_streams():
     def test_pack_64(self):
         # TODO: hex-edit a pack helping us to verify that we can handle 64 byte offsets
         # of course without really needing such a huge pack
-        raise SkipTest()
+        pytest.skip('not implemented')
